universe/ValueRef.cpp

```diff
@@ -932,8 +932,8 @@ int Variable<int>::Eval(const ScriptingContext& context) const
 
     }
     else if (property_name == "LastTurnBattleHere") {
-        if (auto system = std::dynamic_pointer_cast<const System>(object))
-            return system->LastTurnBattleHere();
+        if (auto const_system = std::dynamic_pointer_cast<const System>(object))
```

Comment:
seems unnecessary, but OK...

Reply:
complains of shadow on line 937 `else if (auto system = GetSystem(object->SystemID()))`